server/src/test/java/io/spine/server/delivery/CatchUpTest.java

```diff
@@ -0,0 +1,355 @@
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.delivery;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterators;
+import com.google.common.truth.Truth8;
+import com.google.protobuf.Timestamp;
+import com.google.protobuf.util.Durations;
+import io.spine.base.Time;
+import io.spine.server.DefaultRepository;
+import io.spine.server.delivery.given.ConsecutiveNumberProcess;
+import io.spine.server.delivery.given.ConsecutiveProjection;
+import io.spine.server.delivery.given.CounterCatchUp;
+import io.spine.server.delivery.given.CounterView;
+import io.spine.server.delivery.given.WhatToCatchUp;
+import io.spine.server.entity.Repository;
+import io.spine.test.delivery.ConsecutiveNumberView;
+import io.spine.test.delivery.EmitNextNumber;
+import io.spine.test.delivery.NumberAdded;
+import io.spine.testing.SlowTest;
+import io.spine.testing.server.blackbox.BlackBoxBoundedContext;
+import io.spine.testing.server.blackbox.SingleTenantBlackBoxContext;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import java.util.stream.IntStream;
+
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.protobuf.util.Timestamps.subtract;
+import static io.spine.base.Time.currentTime;
+import static io.spine.server.delivery.CatchUpStatus.COMPLETED;
+import static io.spine.server.delivery.TestRoutines.findView;
+import static io.spine.server.delivery.TestRoutines.post;
+import static io.spine.server.delivery.given.WhatToCatchUp.catchUpAll;
+import static io.spine.server.delivery.given.WhatToCatchUp.catchUpOf;
+import static io.spine.testing.Tests.nullRef;
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toList;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@SlowTest
+@DisplayName("Catch-up of projection instances should")
+public class CatchUpTest extends AbstractDeliveryTest {
+
+    @Override
+    @AfterEach
+    public void tearDown() {
+        super.tearDown();
+        Time.resetProvider();
+    }
+
+    /**
+     * Uses the default wall clock provider, which now emulates the nanoseconds by default.
+     */
+    @Nested
+    @DisplayName(", given the time is provided with nanosecond resolution,")
```

Comment:
Please remove prefixing commas here and below. They look confusing in the test report. It looks as if there was some text before which was accidentally dropped.